### PR TITLE
WIP: Update builder-jobsrv to use PostgreSQL

### PR DESCRIPTION
**Pull request by adamhjk**
This commit switches builder-jobsrv to use PostgreSQL on the backend.

* It uses the new builder-db crate to talk to PostgreSQL
* Follows our denormalized, stored procedure based style
* Includes integration tests for the new jobsrv datastore
* Adds a new state to Jobs, "Dispatched", which is what a job is while
  its in transit to a worker.
* Changes the configuration syntax significantly

_Warning_! Before merging this, be aware that we will no longer be able
to deploy the jobsrv without a running PostgreSQL instance.

Signed-off-by: Adam Jacob <adam@chef.io>

### CODE REVIEW

**Comment by thesentinels** [comment]
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

File: components/builder-jobsrv/tests/integration.rs [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// Copyright (c) 2017 Chef Software Inc. and/or applicable contributors
+// Copyright (c) 2016 Chef Software Inc. and/or applicable contributors
```
Comment (adamhjk) [inline components/builder-jobsrv/tests/integration.rs]: What the fuck, Git?

**Comment by adamhjk** [comment]
@reset these tests are now passing, and this could use a review before we move on.

**Comment by reset** [comment]
This is beautiful

gif-keyboard-6524559332700514335

@thesentinels approve

**Comment by thesentinels** [comment]
:metal: I am testing your branch against master before merging it. We do this to ensure that the master branch is never failing tests.

**Comment by thesentinels** [comment]
:neckbeard: Travis CI has [started testing this PR](https://travis-ci.org/habitat-sh/habitat/builds/197863441).

**Comment by thesentinels** [comment]
:sparkling_heart: Travis CI [reports this PR passed](https://travis-ci.org/habitat-sh/habitat/builds/197863441).

It always makes me feel nice when humans approve of one anothers work. I'm merging this PR now.

I just want you and the contributor to answer me one question:

gif-keyboard-3280869874741411265

**Pull request merged**